test/nettests/whatsapp.cpp

```diff
@@ -0,0 +1,98 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#ifdef ENABLE_INTEGRATION_TESTS
+#define CATCH_CONFIG_MAIN
+#include "private/ext/catch.hpp"
+#include "private/ooni/whatsapp.hpp"
+
+#include "../nettests/utils.hpp"
+
+using namespace mk::nettests;
+using namespace mk;
+
+//TEST_CASE("Synchronous whatsapp test") {
+//    test::nettests::make_test<WhatsappTest>("meek_fronted_requests.txt")
+//        .run();
+//}
+//
+//TEST_CASE("Asynchronous meek-fronted-requests test") {
+//    auto t = test::nettests::make_test<WhatsappTest>("meek_fronted_requests.txt");
+//    test::nettests::run_async(t);
+//}
+
+ TEST_CASE("test net_match") {
+     bool no_err, inclusion;
+     //no_err = mk::ooni::net_match4("10.10.1.44", "10.10.1.32/27", inclusion);
+     //no_err = mk::ooni::net_match4("0.0.0.0", "0.0.0.0/32", inclusion);
+     //no_err = mk::ooni::net_match4("0.0.0.1", "0.0.0.1/24", inclusion);
+     //no_err = mk::ooni::net_match4("0.0.1.0", "0.0.1.0/24", inclusion);
+     //no_err = mk::ooni::net_match4("0.1.0.0", "0.1.0.0/24", inclusion);
+     //no_err = mk::ooni::net_match4("1.0.0.0", "1.0.0.0/24", inclusion);
+     //no_err = mk::ooni::net_match4("1.0.0.0", "1.0.0.0/24", inclusion);
+     //no_err = mk::ooni::net_match4("2000::/3", "2000::/3", inclusion);
+ //    REQUIRE( no_err == true );
+ //    REQUIRE( inclusion == true );
+     //bool b2 = mk::ooni::net_match4("10.10.1.90", "10.10.1.32/27", inclusion);
+ //    REQUIRE( no_err == true );
+ //    REQUIRE( inclusion == false );
+ }
+
+TEST_CASE("ip_to_bytes") {
+    std::vector<uint8_t> ex = { 1, 0, 0, 0 };
+    std::vector<uint8_t> bs = mk::ooni::ip_to_bytes("1.0.0.0");
+    REQUIRE( bs == ex );
+
+    ex = { 255, 255, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+    bs = mk::ooni::ip_to_bytes("ffff::");
+    REQUIRE( bs == ex );
+}
+
+TEST_CASE("same_pre"){
```

Comment:
You can probably _group_ all the `same_pre` tests into the same `TEST_CASE` and use `SECTION` for each specific sub-case so to have logically similar things all together. Also, can you provide more descriptive test cases descriptions?